src/parquet/column/scanner-test.cc

```diff
@@ -0,0 +1,210 @@
+  }
+
+ protected:
+  std::vector<uint8_t> buffer;
+  std::unique_ptr<ColumnDescriptor> descr_;
+  std::vector< std::shared_ptr<Page> > pages_;
+  std::shared_ptr<Scanner> scanner_;
+};
+
+TestPrimitiveScanner::~TestPrimitiveScanner() {}
+
+
+TEST_P(TestPrimitiveScanner, TestBatchSize) {
+  ASSERT_EQ(DEFAULT_SCANNER_BATCH_SIZE, scanner_->batch_size());
+
+  scanner_->SetBatchSize(1);
+  ASSERT_EQ(1, scanner_->batch_size());
+
+  scanner_->SetBatchSize(1000000);
+  ASSERT_EQ(1000000, scanner_->batch_size());
+}
+
+TEST_P(TestPrimitiveScanner, TestSmallBatch) {
+  scanner_->SetBatchSize(1);
+
+  std::stringstream ss;
+  while (scanner_->HasNext()) {
+    scanner_->PrintNext(ss, 1);
+  }
+}
+
+TEST_P(TestPrimitiveScanner, TestPrimitiveColumn) {
+  auto& column_info = GetParam();
+  size_t num_values = std::max(column_info.values.size(), column_info.def_levels.size());
+  std::stringstream ss;
+
+  for (size_t i = 0; i < num_values; i++) {
+    ASSERT_TRUE(scanner_->HasNext());
+    scanner_->PrintNext(ss, column_info.width);
+    ss <<"; ";
+  }
+  ASSERT_FALSE(scanner_->HasNext());
+
+  ASSERT_EQ(column_info.printout, ss.str());
+}
+
+INSTANTIATE_TEST_CASE_P(ScannerTest, TestPrimitiveScanner,
+    testing::Values(
+        // Required column (no definition & repetition levels)
+        PrimitiveColumn(Type::INT32, {1, -2, 3, -4, 5, -6}, {}, 2,
+            "1 ; -2; 3 ; -4; 5 ; -6; "),
+        PrimitiveColumn(Type::INT64, {11, -12, 13, -14, 15}, {}, 3,
+            "11 ; -12; 13 ; -14; 15 ; "),
+        PrimitiveColumn(Type::INT96, {1, 2, 3, 4}, {}, 2,
+            "0 1 1 ; 0 1 2 ; 0 1 3 ; 0 1 4 ; "),
+        PrimitiveColumn(Type::BOOLEAN, {1, 1, 0, 0, 1, 0}, {}, 2,
+            "1 ; 1 ; 0 ; 0 ; 1 ; 0 ; "),
+        PrimitiveColumn(Type::FLOAT, {1234, -567, 89, -1}, {}, 1,
+            "12.340000; -5.670000; 0.890000; -0.010000; "),
+        PrimitiveColumn(Type::DOUBLE, {1234, -567, 89, -1}, {}, 1,
+            "12.340000; -5.670000; 0.890000; -0.010000; "),
+
+        // Optional column (no repetition levels)
+        PrimitiveColumn(Type::INT32, {1, -2, 3, -4, 5}, {1, 0, 1, 1, 0, 0, 1, 1}, 2,
+            "1 ; NULL; -2; 3 ; NULL; NULL; -4; 5 ; "),
+        PrimitiveColumn(Type::INT64, {11, -12, 13, -14, 15}, {1, 0, 1, 1, 0, 0, 1, 1}, 3,
+            "11 ; NULL; -12; 13 ; NULL; NULL; -14; 15 ; "),
+        PrimitiveColumn(Type::INT96, {1, 2, 3, 4}, {1, 0, 1, 1, 0, 0, 1}, 2,
+            "0 1 1 ; NULL; 0 1 2 ; 0 1 3 ; NULL; NULL; 0 1 4 ; "),
+        PrimitiveColumn(Type::BOOLEAN, {1, 1, 0, 0, 1}, {1, 0, 1, 1, 0, 0, 1, 1}, 2,
+            "1 ; NULL; 1 ; 0 ; NULL; NULL; 0 ; 1 ; "),
+        PrimitiveColumn(Type::FLOAT, {1234, -567, 89, -1}, {1, 0, 1, 0, 0, 1, 1}, 1,
+            "12.340000; NULL; -5.670000; NULL; NULL; 0.890000; -0.010000; "),
+        PrimitiveColumn(Type::DOUBLE, {1234, -567, 89, -1}, {1, 0, 1, 1, 0, 0, 1}, 1,
+            "12.340000; NULL; -5.670000; 0.890000; NULL; NULL; -0.010000; ")));
```

Comment:
Testing by examining a text rendering of the output is almost guaranteed to be brittle. Can you refactor to check that values returned with each scan iteration are exactly what was inputted to the test case?